### ISSUE TITLE
Sidekiq support

### ISSUE THREAD

**Issue by benjaminwols** [benngarcia/rails_async_methods#4]
Hi,

We would like to use this gem, but we use Sidekiq which doesn't work at this moment as Sidekiq uses `perform_async` instead of `perform_later`. 
Do you have any plans to support Sidekiq or are you open to a PR which adds Sidekiq support?

**Comment by benngarcia** [comment]
Hey!

This gem is intended to be used with the rails ActiveJob API. This gives you access to a few extra goodies, such as being able to 'drop-in' another backend, the AJ Global ID Serializer, etc. I highly recommend [reconfiguring for ActiveJob usage](https://github.com/mperham/sidekiq/wiki/Active-Job) unless the performance drawbacks are a genuine issue.

<img width="913" alt="image" src="https://user-images.githubusercontent.com/49890553/181110871-40dd6a59-3ce7-4e35-b3a8-05d7923dc709.png">
* Image from link above. Note that the slower speed comes from _pushing_ the jobs to redis. Unless you're reaching throttle speeds where you can't push jobs fast enough, this really shouldn't be an issue. Also it's likely faster with more modern versions 😄 

If that's the case, I am certainly open to a PR which adds Sidekiq support, and happy to help if you'd like to take the lead.
My only preferences would be: 
1) Adding an explicit configuration instead of having these methods implicitly floating around.
2) The dynamic method generation serializes to only the records ID, and passes along the class name as an argument to the `AbstractJob` to be handled at job runtime.

Appreciate your interest! Let me know how I can help.   

Edit: @benjaminwols Just making sure you see this :)

**Comment by benjaminwols** [comment]
Hi @benngarcia 

Sorry for the late reply. We now use this gem in combination with Sidekiq, so no need to support `perform_async` anymore. 

We do however use a feature from Sidekiq-enterprise, namely the unique_for and unique_until options. Would you be open to a PR to add these two options to the `ActiveJobOptionsParser`?

**Comment by benngarcia** [comment]
Hi @benjaminwols 

Definitely open to a PR. Instead of adding the options directly to the `ActiveJobOptionsParser`, I would prefer to define an additional module that can be optionally mixed-in to the `ActiveJobOptionsParser` class (with a config block, preferably).

But, I do recognize that for your use case you don't need to support other mixins. If you're able to just make it work for your use case and add the necessary tests, I'll get to my refactoring likely within a week and put out a new minor release.

Thank you!

**Comment by benjaminwols** [comment]
Hi @benngarcia 

I tried to implement the `unique_for` and `unique_until` options, but I found that the arguments are not passed from AJ to Sidekiq. I found this issue in the Sidekiq gem [#4281](https://github.com/sidekiq/sidekiq/issues/4281), which tells that there is no way to pass these arguments and that the only way is to use a native `Sidekiq::Job`.

I will close this issue as I don't see a way to support these options in combination with AJ. It would be nice if you are able to put out a bugfix release for the `:wait` option bug.
